CorsixTH/Src/th_sound.cpp

```diff
@@ -150,179 +150,179 @@ size_t THSoundArchive::getSoundDuration(size_t iIndex)
 
 #undef FOURCC
 
-SDL_RWops* THSoundArchive::loadSound(size_t iIndex)
+SDL_RWops* th_sound_archive::load_sound(size_t iIndex)
 {
-    if(iIndex >= m_iFileCount)
+    if(iIndex >= sound_file_count)
         return nullptr;
 
-    th_fileinfo_t *pFile = m_pFiles + iIndex;
-    return SDL_RWFromConstMem(m_pData + pFile->position, pFile->length);
+    th_fileinfo *pFile = sound_files + iIndex;
+    return SDL_RWFromConstMem(data + pFile->position, pFile->length);
 }
 
 #ifdef CORSIX_TH_USE_SDL_MIXER
 
-THSoundEffects* THSoundEffects::ms_pSingleton = nullptr;
+sound_player* sound_player::singleton = nullptr;
 
-THSoundEffects::THSoundEffects()
+sound_player::sound_player()
 {
-    m_ppSounds = nullptr;
-    m_iSoundCount = 0;
-    ms_pSingleton = this;
-    m_iCameraX = 0;
-    m_iCameraY = 0;
-    m_fCameraRadius = 1.0;
-    m_fMasterVolume = 1.0;
-    m_fSoundEffectsVolume = 0.5;
-    m_iPostionlessVolume = MIX_MAX_VOLUME;
-    m_bSoundEffectsOn = true;
+    sounds = nullptr;
+    sound_count = 0;
+    singleton = this;
+    camera_x = 0;
+    camera_y = 0;
+    camera_radius = 1.0;
+    master_volume = 1.0;
+    sound_effect_volume = 0.5;
+    positionless_volume = MIX_MAX_VOLUME;
+    sound_effects_enabled = true;
 
 #define NUM_CHANNELS 32
 #if NUM_CHANNELS >= 32
-    m_iChannelStatus = ~0;
+    available_channels_bitmap = ~0;
     Mix_AllocateChannels(32);
 #else
-    m_iChannelStatus = (1 << NUM_CHANNELS) - 1;
+    channels_in_use_bitmap = (1 << NUM_CHANNELS) - 1;
     Mix_AllocateChannels(NUM_CHANNELS);
 #endif
 #undef NUM_CHANNELS
 
-    Mix_ChannelFinished(_onChannelFinish);
+    Mix_ChannelFinished(on_channel_finished);
 }
 
-THSoundEffects::~THSoundEffects()
+sound_player::~sound_player()
 {
-    setSoundArchive(nullptr);
-    if(ms_pSingleton == this)
-        ms_pSingleton = nullptr;
+    populate_from(nullptr);
+    if(singleton == this)
+        singleton = nullptr;
 }
 
-void THSoundEffects::_onChannelFinish(int iChannel)
+void sound_player::on_channel_finished(int iChannel)
 {
-    THSoundEffects *pThis = getSingleton();
+    sound_player *pThis = get_singleton();
     if(pThis == nullptr)
         return;
 
-    pThis->releaseChannel(iChannel);
+    pThis->release_channel(iChannel);
 }
 
-THSoundEffects* THSoundEffects::getSingleton()
+sound_player* sound_player::get_singleton()
 {
-    return ms_pSingleton;
+    return singleton;
 }
 
-void THSoundEffects::setSoundArchive(THSoundArchive *pArchive)
+void sound_player::populate_from(th_sound_archive *pArchive)
 {
-    for(size_t i = 0; i < m_iSoundCount; ++i)
+    for(size_t i = 0; i < sound_count; ++i)
```

Comment:
looks hopelessly silly, the archive should know the number of sounds, or it should be provided as parameter